utils/graph.c

```diff
@@ -172,6 +188,13 @@ static int add_graph_exit(struct uftrace_task_graph *tg)
 out:
 	node->total_time.sum += fstack->total_time;
 	node->child_time += fstack->child_time;
+	node->self_time.sum += fstack->total_time - fstack->child_time;
+
+	update_graph_time_stat(&node->total_time, fstack->total_time);
+	update_graph_time_stat(&node->self_time, fstack->total_time - fstack->child_time);
+
+	node->total_time.avg = (node->total_time.sum) / node->nr_calls;
+	node->self_time.avg = (node->self_time.sum) / node->nr_calls;
```

Comment:
I hit divide by zero error here because `node->nr_calls` comes with `0` in this case. This can be reproduced as follows.
```
$ gcc -pg -o t-fork tests/s-fork.c
$ uftrace record t-fork
$ uftrace tui -f +total-avg,total-min,total-max,self-avg,self-min,self-max
```
If you add the following line above, you will hit the assertion failure easily.
```c
ASSERT(node->nr_calls > 0);
```

Reply:
For those cases, ASSERT will crash the uftrace and I don't think we want it.  Maybe we can drop the node or set nr_calls to 1.

Reply:
@namhyung 
> Maybe we can drop the node or set nr_calls to 1.

I don't know if this is what you mean.

```
if (node->nr_calls > 0) {
		node->total_time.avg = (node->total_time.sum) / node->nr_calls;
		node->self_time.avg = (node->self_time.sum) / node->nr_calls;
}
```

Reply:
I thought it's an error if `node->nr_calls != 1`.